### PR TITLE
Import old utemplates

### PR DESCRIPTION
**Pull request by kordan**
Up to now usertemplates built with old version of the module were refused from the last version of surveypro.
Furthermore, plugin version was used to detect if item plugins were old and this is not a good strategy.
As far as I know, this patch allow old usertemplate and mastertemplate both.

### CODE REVIEW

**Comment by stronk7** [comment]
Ah, seems you had based this branch in the item_setup_form_missing_defaults (#41) one. But the last rebase was on master. So now the first commit here is not correct, because has been already merged by the other PR with different commit hash.

So, there are 2 options... Or you rebase -i again with master (now than the other PR has been merged) and get rid of the duplicate commit... or I cherry pick manually your second commit and apply it, instead of merging (because merging will add the 2 commits).

As you wish... in the mean time... I'm looking to the patch.

**Comment by stronk7** [comment]
It's big patch, but changes look ok. So just tell me if you amend the branch (getting rid of dupe commit) or I pick the correct commit and done.

**Comment by kordan** [comment]
This is what I did:

[daniele@/Applications/MAMP/htdocs/head/mod/surveypro (master $=)]$ git co import_old_templates 
Switched to branch 'import_old_templates'
[daniele@/Applications/MAMP/htdocs/head/mod/surveypro (import_old_templates $)]$ git rebase master
First, rewinding head to replay your work on top of it...
Applying: CONTRIB-6167 mod_surveypro: to import and apply old usertemplates'

and it seems all is fine.
Strange that:
git diff remotes/origin/import_old_utemplates
is empty. Any suggestion?

File: classes/itembase.class.php [hunk 1]
```diff
@@ -850,6 +850,21 @@ public function item_mandatory_is_allowed() {
     }
 
     /**
+     * item_add_mandatory_base_fields
+     * Copy mandatory fields to $record.
+     *
+     * @param stdClass $record
+     * @return nothing
```
Comment (stronk7) [inline classes/itembase.class.php]: here you can use `@return void` or no return phpdoc, as you prefer. But those `@return nothing` are not correct.

Comment (kordan) [inline classes/itembase.class.php]: It has already been done in the next branch

File: utemplates_apply.php [hunk 2]
```diff
@@ -73,9 +73,9 @@
 // Begin of: manage form submission.
 $utemplateman->formdata = $applyutemplate->get_data();
 if ($utemplateman->formdata || ($confirm == SURVEYPRO_CONFIRMED_YES)) {
-    // Here I don't need to execute validate_xml because xml was validated at upload time
-    // Here I only need to verfy that plugin versions still match
-    $utemplateman->check_items_versions();
+    // Here I don't need to execute validate_xml because xml was validated at upload time.
+    // Here I only need to verfy that plugin versions still match.
+    // $utemplateman->check_items_versions();
```
Comment (stronk7) [inline utemplates_apply.php]: This is only an example. I've seen that you here and there use to keep old code commented. One line or multiple lines. That's not good in general. If you've to explain something do it, but not keeping old code. In fact I bet the codechecker will, probably cry about it.

Comment (kordan) [inline utemplates_apply.php]: True! Codechecker crys a lot.
I will! :(

**Comment by stronk7** [comment]
Merged, thanks!

**Pull request merged**